src/algebra/fields/bigint.tcc

```diff
@@ -164,6 +165,78 @@ bool bigint<n>::test_bit(const std::size_t bitno) const
     }
 }
 
+template<mp_size_t n> template<mp_size_t m>
+inline bigint<n> bigint<n>::operator+(const bigint<m>& other) const
+{
+    static_assert(n >= m, "first arg must not be smaller than second arg for bigint add");
+    bigint<n> res;
+    mpn_add(res.data, data, n, other.data, m);
+    return res;
+}
+
+template<mp_size_t n> template<mp_size_t m>
+inline void bigint<n>::operator+=(const bigint<m>& other)
+{
+    static_assert(n >= m, "first arg must not be smaller than second arg for bigint in-place add");
+    mpn_add(data, data, n, other.data, m);
+}
+
+template<mp_size_t n> template<mp_size_t m>
+inline bigint<n+m> bigint<n>::operator*(const bigint<m>& other) const
+{
+    static_assert(n >= m, "first arg must not be smaller than second arg for bigint mul");
```

Comment:
See the documentation of mpn_mul at https://gmplib.org/manual/Low_002dlevel-Functions.html . I thought it was weird too, but didn't want to add the complexity of swapping the arguments when necessary.